docs/book/src/developer/providers/v1.0-to-v1.1.md

```diff
@@ -13,6 +13,8 @@ maintainer of other providers and consumers of our Go API.
 are kept in sync with the versions used by `sigs.k8s.io/controller-runtime`. 
 
 - sigs.k8s.io/controller-runtime: v0.10.x => v0.11.x
+- github.com/go-logr/logr: v0.4.0 => v1.2.0
```

Comment:
I want to mention those two dependencies this time as the go-logr change is pretty much breaking.

Reply:
Q: (not strictly related to this line but related to this paragraph)
should we document which version of Kubernetes we are importing as a "derived from controller runtime"?